utils/schema_parser.py

```
@@ -946,3 +1050,318 @@ def _validate_value_type(self, value: Any, expected_type: Any, current_path: str
+            entity_data: The entity's data dict
+            entity_type: Entity type name
+            path: Dot-separated path (e.g., "group_status.Rebel Force.position")
+            default: Default value if not found
+
+        Returns:
+            Value at path or default
+        """
+        parts = path.split(".")
+        current = entity_data
+
+        for i, part in enumerate(parts):
```

Ruff (0.15.12)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)